core/src/main/java/org/geysermc/geyser/configuration/GeyserConfiguration.java

```diff
@@ -160,6 +162,10 @@
         String getUniqueId();
     }
 
+    interface IMsaAuthenticationConfiguration {
```

Comment:
I thought it would be pretty short-sighted to introduce single `pendingMicrosoftAuthenticationTimeout` config entry. If someone would continue working on MS authentication, they would add their configurable entries here.

Reply:
Personally, I would continue under IUserAuthenticationInfo - eventually only Microsoft accounts will be supported (as Mojang will drop Mojang account support) so there's no need for a Microsoft-specific section.

Reply:
Timeout configuration has to be configured globally. As far as I know `userAuths` is for per-user configuration.

Reply:
Good point. What other settings would you envision being in this section? I don't think we need to label it as the Microsoft authentication section because that'll be counterintuitive in the future.

Reply:
I believe it would be a good fit for things like: token (default) storage time, OAuth configuration (Azure app client id & secret, redirect URL)...

I'm fine calling it any way as long as it's technically correct :)

Reply:
I don't think we need another configuration section for that - all our ping passthrough options aren't in their own config, for example. I will inquire from others since this is incredibly nitpicky.

Reply:
No problem! Let me know what you'll decide.

Reply:
I don't think we need the section. If we just keep the settings in their own block in the YAML, that should be clear enough, in my opinion. The setting can still be called `pending-authentication-timeout`. It may actually fit fine in the Remote section just because we have Allow Password Authentication already in there, but that setting may be removed in the future so we don't have to follow precedent.

Reply:
👌